platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py

```diff
@@ -79,6 +87,13 @@ def _load_eeprom(self):
             except Exception as e:
                 logger.log_error('Failed to remove cache file {} - {}'.format(cache_file, repr(e)))
 
+        if 'simx' in self._onie_platform:
+            platform_path = get_path_to_platform_dir()
+            
+            if not os.path.exists(platform_path):
+                platform_path = '/usr/share/sonic/platform'
```

Comment:
@tomer-israel why do we need this?

Reply:
I remember that we looked at the code together and decided that this is needed.
when i'm running without this the decode-syseeprom is failing

Reply:
@tomer-israel `get_path_to_platform_dir()` should handle it. If no - we need to fix API.

**Example:**
https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-py-common/sonic_py_common/device_info.py#L160
```python
def get_path_to_platform_dir():
    """
    Retreives the paths to the device's platform directory
    Returns:
        A string containing the path to the platform directory of the device
    """
    # Get platform
    platform = get_platform()

    # Determine whether we're running in a container or on the host
    platform_path_host = os.path.join(HOST_DEVICE_PATH, platform)

    if os.path.isdir(CONTAINER_PLATFORM_PATH):
        platform_path = CONTAINER_PLATFORM_PATH
    elif os.path.isdir(platform_path_host):
        platform_path = platform_path_host
    else:
        raise OSError("Failed to locate platform directory")

    return platform_path
```